### ISSUE TITLE
zimsplit (zim-tools_win-x86_64-3.5.0) can not handle well zim.files biger than ca.80gb spliting to Default: 2GB

### ISSUE THREAD

**Issue by BD-RE** [openzim/zim-tools#433]
I tried to split wikipedia_en_all_maxi_2024-01.zim with 102 GB (109.885.670.576 Bytes) to Default: 2GB.
Result I got ca. 40 times 2GB files and a bigger leftover. 

-------
zim-tools 3.5.0

libzim 9.2.3
+ libzstd 1.5.5
+ liblzma 5.2.11
+ libxapian 1.4.23
+ libicu 74.1.0

**Comment by BD-RE** [comment]
C:\Users\me>"C:\Users\me\Desktop\offline-wiki burn-projekt\Tools\zim-tools_win-x86_64-3.5.0\zimsplit.exe"  "y:\kiwix-zim\wikipedia_en_all_maxi_2024-01.zim"
opening new file y:\kiwix-zim\wikipedia_en_all_maxi_2024-01.zimaa
opening new file y:\kiwix-zim\wikipedia_en_all_maxi_2024-01.zimab
opening new file y:\kiwix-zim\wikipedia_en_all_maxi_2024-01.zimac
opening new file y:\kiwix-zim\wikipedia_en_all_maxi_2024-01.zimad
opening new file y:\kiwix-zim\wikipedia_en_all_maxi_2024-01.zimae
........
log shortened see the last 4 lines:
........
opening new file y:\kiwix-zim\wikipedia_en_all_maxi_2024-01.zimbl
opening new file y:\kiwix-zim\wikipedia_en_all_maxi_2024-01.zimbm
opening new file y:\kiwix-zim\wikipedia_en_all_maxi_2024-01.zimbn
opening new file y:\kiwix-zim\wikipedia_en_all_maxi_2024-01.zimbo
opening new file y:\kiwix-zim\wikipedia_en_all_maxi_2024-01.zimbp
WARNING: Part y:\kiwix-zim\wikipedia_en_all_maxi_2024-01.zimbp is bigger that max part size. (3485777929>2147483648)
opening new file y:\kiwix-zim\wikipedia_en_all_maxi_2024-01.zimbq
opening new file y:\kiwix-zim\wikipedia_en_all_maxi_2024-01.zimbr
WARNING: Part y:\kiwix-zim\wikipedia_en_all_maxi_2024-01.zimbr is bigger that max part size. (19317566780>2147483648)
..........
.........dir shortened see the last 4 lines:

12.11.2024  13:30     2.146.437.678 wikipedia_en_all_maxi_2024-01.zimbk
12.11.2024  13:32     2.147.268.773 wikipedia_en_all_maxi_2024-01.zimbl
12.11.2024  13:35     2.146.431.290 wikipedia_en_all_maxi_2024-01.zimbm
12.11.2024  13:37     2.146.400.141 wikipedia_en_all_maxi_2024-01.zimbn
12.11.2024  13:39     1.206.591.345 wikipedia_en_all_maxi_2024-01.zimbo
12.11.2024  13:43     3.485.777.929 wikipedia_en_all_maxi_2024-01.zimbp
12.11.2024  13:43            93.999 wikipedia_en_all_maxi_2024-01.zimbq
12.11.2024  14:06    19.317.566.780 wikipedia_en_all_maxi_2024-01.zimbr

**Comment by kelson42** [comment]
Yes, here there is nothing we can do as we can not cut the fulltext index...

**Comment by BD-RE** [comment]
Did I understand well, the full text index is appended at the end of the archive and in this case ca. 19gb?
But in this case (status) it is functional.
If somebody split the left over of the process anyway, he have to join it together to get it functional.

**Comment by veloman-yunkan** [comment]
`zimsplit` is not a general purpose file splitting utility. It cuts a ZIM file only at points which allow the readers to work with such a split ZIM file without needing to reassemble the pieces into one whole file.
